packages/integration-tests/test/actions.spec.ts

```diff
@@ -65,27 +72,113 @@ const deposit = async (amount, value) => {
   const receipt = await watcher.getL2TransactionReceipt(msgHash)
 }
 
-describe('Messages', async () => {
+const withdraw = async (value) => {
+  const L2Messenger = new Contract(l2MessengerAddress, l2MessengerJSON.interface, l2Wallet)
+  const calldata = simpleStorage.interface.encodeFunctionData('setValue', [value])
+  const l2ToL1Tx = await L2Messenger.sendMessage(
+    simpleStorage.address,
+    calldata,
+    5000000,
+    { gasLimit: 7000000 }
+  )
+  await l2Provider.waitForTransaction(l2ToL1Tx.hash)
+  const count = (await simpleStorage.totalCount()).toString()
+  await sleep(60000)
+}
+
+describe('SimpleStorage', async () => {
+  before(async () => {
+    const web3 = new Web3Provider(
+      ganache.provider({
+        mnemonic: Config.Mnemonic(),
+      })
+    )
+  })
 
   it('should initialize the watcher', async () => {
     watcher = await initWatcher()
   })
 
   it('should deploy the simple storage contract', async () => {
-    SimpleStorage = await deploySimpleStorage()
+    simpleStorage = await simpleStorageFactory.deploy()
+    await simpleStorage.deployTransaction.wait()
   })
 
   it('should deposit from L1->L2', async () => {
     const value = `0x${'42'.repeat(32)}`
     await deposit(1, value)
-    const msgSender = await SimpleStorage.msgSender()
-    const l1ToL2Sender = await SimpleStorage.l1ToL2Sender()
-    const storageVal = await SimpleStorage.value()
-    const count = await SimpleStorage.totalCount()
+    const msgSender = await simpleStorage.msgSender()
+    const l1ToL2Sender = await simpleStorage.l1ToL2Sender()
+    const storageVal = await simpleStorage.value()
+    const count = await simpleStorage.totalCount()
 
     msgSender.should.be.eq(l2MessengerAddress)
     l1ToL2Sender.should.be.eq(l1Wallet.address)
     storageVal.should.be.eq(value)
     count.toNumber().should.be.eq(1)
   })
+
+  // TODO: Enable withdrawal test
+  it.skip('should withdraw from L2->L1', async () => {
+    const value = `0x${'77'.repeat(32)}`
+    await withdraw(value)
+    const msgSender = await simpleStorage.msgSender()
+    const l2ToL1Sender = await simpleStorage.l2ToL1Sender()
+    const storageVal = await simpleStorage.value()
+    const count = await simpleStorage.totalCount()
+
+    msgSender.should.be.eq(l1MessengerAddress)
+    l2ToL1Sender.should.be.eq(l2Wallet.address)
+    storageVal.should.be.eq(value)
+    count.toNumber().should.be.eq(1)
+  })
+})
+
+describe('ERC20', async () => {
+  const alice = new Wallet(etherbase, l2Provider)
```

Comment:
we sure this works? Just checked and it looks like etherbase is an address, not a private key?

Reply:
It does work the theory being that this is one of the default addresses so maybe the pk is queryable? We could try using a different new test pk too. Do you guys have one in mind that is not the deployer pk?

Reply:
It should be safe to use the deployer private key here since there won't be a race condition. I'm a bit confused as well how this works if you pass in an address, it looks like this `Wallet` is used to deploy the erc20 contract